src/FunctionRef.h

```diff
@@ -0,0 +1,62 @@
+
+#pragma once
+
+// Guard against macro new
+#pragma push_macro("new")
+#undef new
+
+// Declared only so it can be partially specialized
+template <class Signature>
+class cFunctionRef;
+
+/** A light-weight, type-erased reference to a function object.
+Non-owning, be mindful of lifetimes. */
+template <class Ret, class... Args>
+class cFunctionRef<Ret(Args...)>
```

Comment:
Why exactly do we need a new object for this? Shouldn't using `std::function` suffice? What is the use-case that forces us to use this?

Reply:
Can you elaborate with an example? I still don't get it :(

Reply:
Sorry but which part don't you get?  Do you mean why `std::function` might allocate memory?

Reply:
I'd like an example for which you felt `std::function` was not good enough (or unusable) and which the new framework fixes.

Reply:
It's mainly for performance reasons.  Similar to `std::string` vs `std::string_view`.
I can leave it out and use `std::function` if you'd prefer.

Reply:
It seems like a premature optimization to me. Let's go with `std::function` for now and if it proves to be a bottleneck in the future, we can revisit this framework.

Reply:
Categorising 50 lines of code as a "framework" seems a bit extreme.

Reply:
Yeah, but it's dense code, quite difficult to understand what it actually does :) Which is exactly the definition of framework (or library - ever looked into STL? :)

Reply:
I could add more documentation in `cFunctionRef` if that helps.  The principle is very simple and it isn't doing much really.

It's not much different from the c-style stuff used in the network code.